### DISCUSSION TITLE
GridmapNavSimul - Gridmap not showing

### DISCUSSION THREAD

**Discussion by Paul-Hay** [MRPT/mrpt#1326] [Q&A] [answered]
I am running mrpt on Windows. 
When I load my gridmap, the map loads and I get the notification that, but I default gridmap does not change to my new map. The sensor range detects objects from my new map, but the map does not show. The default map is stuck on my screen. 
I have tried maps in png, pgm, bmp formats and I still get same results. I have also experienced same issue trying to run on my Ubuntu 20.04.
Can I get help with this issue?

**Comment by jlblancoc** [comment, marked as answer]
Hi, 
It turned out to be a regression introduced in a recent MRPT/mrpt version. It has just been fixed with this commit: https://github.com/MRPT/mrpt/commit/16769a3552a661e02ff7259bbf8b56aefe26a270

It will come fixed in v2.14.5. In the meanwhile, you may need to build from sources to have a version without that bug (!). 

You can also consider using MVSim, which allows similar 2D world simulations (plus much more).

**Reply by Paul-Hay** [reply]
Can you help me set up mvsim? I am running Ubuntu, and I am getting the following error when I install using the command [ sudo apt install ros-$ROS_DISTRO-mvsim ] and run the first example code [ ros2 launch mvsim demo_warehouse.launch.py ]. My rviz window remains blank.


![shot](https://github.com/user-attachments/assets/56e6e0cd-bc26-491c-8806-4623cc4e0eba)

I also get the following error:

[rviz2-2] [ERROR] [1731304308.308577251] [rviz2]: PluginlibFactory: The plugin for class 'rviz_plugin_tutorials/Teleop' failed to load. Error: According to the loaded plugin descriptions the class rviz_plugin_tutorials/Teleop with base class type rviz_common::Panel does not exist. Declared types are 

[ERROR] [mvsim_node-1]: process has died [pid 4472, exit code 1, cmd '/opt/ros/jazzy/lib/mvsim/mvsim_node --ros-args -r __node:=mvsim --params-file /opt/ros/jazzy/share/mvsim/mvsim_tutorial/mvsim_ros2_params.yaml --params-file /tmp/launch_params_865l6bcn'].

I checked the path and the file mvsim_ros2_params.yaml file exist.

**Reply by jlblancoc** [reply]
Before the " process has died" message, look a bit up, there must be some other error message explaining what happened to mvsim so it exited... Look for "exception", it may help.
